include/aws/common/hash_table.h

```diff
@@ -74,18 +75,30 @@ struct aws_hash_element {
     void *value;
 };
 
+enum aws_hash_iter_status {
+    AWS_HASH_ITER_STATUS_DONE,
+    AWS_HASH_ITER_STATUS_DELETE_CALLED,
+    AWS_HASH_ITER_STATUS_READY_FOR_USE,
+};
+
+/* aws_hash_iter has reserved fields of type void*. If this assertion below is true, we can
+ * use intptr_t in these fields without issue.
+ */
+AWS_STATIC_ASSERT(sizeof(intptr_t) == sizeof(void *));
+
 struct aws_hash_iter {
     const struct aws_hash_table *map;
     struct aws_hash_element element;
     size_t slot;
     size_t limit;
+    /* Holds an enum aws_hash_iter_status */
+    intptr_t status;
```

Comment:
enums are ints underneath, you could leave an int reserved and only use an int for status

Reply:
Done.